sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ManagementChannel.java

```diff
@@ -59,9 +62,17 @@
     private final MessageSerializer messageSerializer;
     private final TokenManagerProvider tokenManagerProvider;
 
+    /*This is to maintain cbs node and get authorization.*/
+    private final TokenManager cbsBasedTokenManager;
+
+    // Maintain last peek sequence number
+    private AtomicReference<Long>  lastPeekedSequenceNumber = new AtomicReference<>(0L);
+
+
     ManagementChannel(Mono<RequestResponseChannel> responseChannelMono, String topicOrQueueName,
                       TokenCredential credential, TokenManagerProvider tokenManagerProvider,
-                      MessageSerializer messageSerializer, Scheduler scheduler) {
+                      MessageSerializer messageSerializer, Scheduler scheduler, TokenManager cbsBasedTokenManager
```

Comment:
Why do we pass in the TokenManager here. iirc, before we create the ManagementChannel, we already create a token manager that authorizes with it. We shouldn't need to manually authorize ourselves.

Reply:
The `TokenManager` instance is created before `ManagementChannel` is created but does not call `TokenManager.authorize()`. So we pass this instance in `ManagementChannel`  because  it knows when user makes first call to `peek`. First call to `authorize()` should happen when user does peek operation, So  we call it `once only` just before peek is called.